lib/modules/datasource/docker/index.ts

```diff
@@ -854,8 +854,16 @@ export class DockerDatasource extends Datasource {
           foundMaxResultsError = true;
           continue;
         }
+
         throw err;
       }
+
+      if (res.headers['x-artifactory-id'] && url.includes('n=')) {
```

Comment:
@rarkins I'm thinking about how we can detect the different hosts at token fetch and cache that info, so we can start with right page size with first request. 🤔

Reply:
OK so an extra query every time we encounter a host for the first time on a run? Also need to consider that it's likely that such a host would have multiple packages and therefore we'd likely be looking them up in parallel, so we'd want to try to optimise so that we don't end up resolving the host type redundantly in parallel too (which could happen in today's solution too).

Reply:
i think we can reuse the request to `/v2` for auth check and remember the returned headers.

Reply:
Great idea

Reply:
Thanks, folks, for the feedback. Please tell me where I could refer to the "storing in-memory" part.